include/libtorrent/bencode.hpp

```diff
@@ -88,6 +88,28 @@ namespace libtorrent
 
 	namespace detail
 	{
+		template <typename T, typename Cond = typename std::enable_if<
```

Comment:
but.. making it a template makes it implicitly inline, and since you're constraining the template like this, being inline is really the only property that's left. The fact that it's inline now is why your patch works. and, yes. I wish more of the bdecode code could be moved into the library (which is why I put this function in there).